### DISCUSSION TITLE
Create external navigation link

### DISCUSSION THREAD

**Discussion by wonderplayer** [pnp/pnpcore#610] [Q&A] [answered]
Hello,

I'm struggling with creating external navigation link through SharePoint REST API. 

POST Request URL: https://contoso.sharepoint.com/_api/web/navigation/TopNavigationBar
Body:
``` JSON
{
    "__metadata": {
        "type": "SP.NavigationNode"
    },
    "Url": "https://contoso.sharepoint.com/nonexisting",
    "Title": "Nonexisting",
    "IsExternal": true
}
```

Error response:
``` JSON
{
    "odata.error": {
        "code": "-2130247147, Microsoft.SharePoint.SPException",
        "message": {
            "lang": "en-US",
            "value": "Cannot open \"/nonexisting\": no such file or folder."
        }
    }
}
```

I see that in PnP.PowerShell there is an option to mark navigation node as external, and it does solve the issue. Took a peek in hot it is implemented [there](https://github.com/pnp/powershell/blob/dev/src/Commands/Navigation/AddNavigationNode.cs) - seems that with CSOM it just works, but with REST API this property is getting ignored or something.
This might be a potential issue, when implementing navigation model for PnPCore.

Maybe somebody knows how to do it? Even posted a question in [Stack exchange](https://sharepoint.stackexchange.com/questions/298639/create-navigation-item-using-rest-api-with-non-existing-link-in-sharepoint-onlin) 😄

Thank you!

**Comment by jansenbe** [comment, marked as answer]
Hi @wonderplayer ,

Your code works for me:

![image](https://user-images.githubusercontent.com/7451219/140296824-9470a7e8-11e2-43d1-ae62-3a1a054850a6.png)

**Reply by wonderplayer** [reply]
Hi @jansenbe 
Yes, if it is a completely external link.
But try to change it like this: https://\<tenant\>.sharepoint.com/sites/nonexisitng.
And execute it against root site.

In your example you could put URL: https://bertonline.sharepoint.com/sites/prov-2/nonexisting

**Reply by jansenbe** [reply]
@wonderplayer : managed to reproduce your issue and did not immediately have a fix for this, seems that for the REST call additional checks on the URL are done whenever we detect the URL is for the current site. What does work is adding the nav entry like I showed above, followed by updating it with the non existing url inside the current site:

![image](https://user-images.githubusercontent.com/7451219/140304349-5647e359-bb36-49cf-a30e-03d6761fd3ba.png)

And in SP:

![image](https://user-images.githubusercontent.com/7451219/140304461-4677e959-ef70-40c5-9487-1d0273cff2c4.png)

**Reply by wonderplayer** [reply]
My days would be so much darker without you @jansenbe 
Thank you!

P.S. Guess this will be the way in PnPCore as well.
